### PR TITLE
Add guide to add new libdatadog features.

### PR DESCRIPTION
**Pull request by hoolioh**
### Description

Provide documentation to integrate new features from libdatadog.

### CODE REVIEW

File: LIBDATADOG.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Libdatadog
+Common components are integrated in dd-tracer-php by importing the source code from libdatadog repository. As stated in
```
Comment (Anilm3) [inline LIBDATADOG.md:2]:
```suggestion
Common components are integrated in dd-trace-php by importing the source code from the libdatadog repository. As stated in
```

File: LIBDATADOG.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+The rust sources are compiled to assemble a static library named libddtrace_php.a, this library contains the necessary
+functionality to:
+* Sidecar initialization and communication.
+* Sending telemetry.
+* Common functionality like container identification and runtime id generation.
+
+The compilation process is handled by a [compile script](https://github.com/DataDog/dd-trace-php/compile_rust.sh) which
+is instantiated from the main Makefile.
+Once the library is assembled its contents are linked in ddtrace.so which is the main objective in the Makefile. This
+library is the extension which the PHP engine will load on runtime in order to provide all tracing functionality.
```
Comment (Anilm3) [inline LIBDATADOG.md:15]:
```suggestion
library is the extension which the PHP engine will load at runtime in order to provide all tracing functionality.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -11,6 +11,8 @@ git submodule init
 git submodule update
 ```
 
+To integrate new features from datadog shared library please refer to this [guide](https://github.com/DataDog/dd-trace-php/LIBDATADOG.md).
```
Comment (Anilm3) [inline CONTRIBUTING.md:14]:
```suggestion
To integrate new tracing and profiling features from the datadog shared library please refer to this [guide](https://github.com/DataDog/dd-trace-php/LIBDATADOG.md).
```

**Comment by bwoebi** [comment]
The links are broken, and need `tree/master/` after the repository name to work.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -11,6 +11,8 @@ git submodule init
 git submodule update
 ```
 
+To integrate new tracing and profiling features from the datadog shared library please refer to this [guide](https://github.com/DataDog/dd-trace-php/LIBDATADOG.md).
```
Comment (bwoebi) [inline CONTRIBUTING.md:14]:
```suggestion
To integrate new tracing and profiling features from the datadog shared library please refer to this [guide](https://github.com/DataDog/dd-trace-php/tree/master/LIBDATADOG.md).
```

File: LIBDATADOG.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Libdatadog
+Common components are integrated in dd-trace-php by importing the source code from the libdatadog repository. As stated in
+the [contributing file](https://github.com/DataDog/dd-trace-php/CONTRIBUTING.md) the code is imported by using git
```
Comment (bwoebi) [inline LIBDATADOG.md:3]:
```suggestion
the [contributing file](https://github.com/DataDog/dd-trace-php/blob/master/CONTRIBUTING.md) the code is imported by using git
```

File: LIBDATADOG.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+Once the library is assembled its contents are linked in ddtrace.so which is the main objective in the Makefile. This
+library is the extension which the PHP engine will load at runtime in order to provide all tracing functionality.
+
+## Adding new features
+Upon adding new modules in the libdatadog repository there is the need to create the neccessary glue code in the php
+side to use those new modules. For that purpose you'll need to follow the next steps:
+* Modify the [project file in components-rs](https://github.com/DataDog/dd-trace-php/blob/master/components-rs/Cargo.toml) to add
+the new dependency.
+* Add the necessary code in components-rs to expose the functionality.
+* Modify the compile script if any new configuration needs to be added to the compilation process.
```
Comment (bwoebi) [inline LIBDATADOG.md:23]:
```suggestion
* Modify the compile script if any new configuration needs to be added to the compilation process.
* Add the new module to the `RUST_FILES` variable in the [Makefile](https://github.com/DataDog/dd-trace-php/blob/master/Makefile).
```

**Comment by bwoebi** [comment]
Added a couple suggestions, I think we can merge it afterwards.

**Review by bwoebi — APPROVED** [review]

**Pull request merged**